### PR TITLE
Tests for Paginator

### PR DESCRIPTION
**Pull request by satyasekharcvb**
### What does this PR do?
Adds tests for the paginator component

## The PR fulfills these requirements:

[X ] Tests for the proposed changes have been added/updated.
[ X] Code linting and formatting was performed.

### Functionality Before

NA

### Functionality After

NA

### CODE REVIEW

File: force-app/main/default/lwc/paginator/__tests__/paginator.test.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        //Set the public property values
+        element.pageNumber = 0;
+        element.pageSize = 9;
+        element.totalItemCount = 0;
+
+        document.body.appendChild(element);
+
+        // Query div for validating the display message on component init
+        const lightningLayoutItemEl = element.shadowRoot.querySelector(
+            'lightning-layout-item.nav-info'
```
Comment (pozil) [inline force-app/main/default/lwc/paginator/__tests__/paginator.test.js:56]: simplify selector to just `.nav-info`

File: force-app/main/default/lwc/paginator/__tests__/paginator.test.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+        // Mock handlers for child events
+        const handlerPrevious = jest.fn();
+        const handlerNext = jest.fn();
+        // Add event listener to catch child events
+        element.addEventListener('previous', handlerPrevious);
+        element.addEventListener('next', handlerNext);
+
+        element.shadowRoot
+            .querySelectorAll('lightning-button-icon')
```
Comment (pozil) [inline force-app/main/default/lwc/paginator/__tests__/paginator.test.js:27]: BTW this is not a good way to test: clicking all buttons at once won't tell you if previous does indeed go back.
Test would still pass if button were doing the opposite of what label says.

File: force-app/main/default/lwc/paginator/__tests__/paginator.test.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+describe('c-paginator', () => {
+    afterEach(() => {
+        // The jsdom instance is shared across test cases in a single file so reset the DOM
+        while (document.body.firstChild) {
+            document.body.removeChild(document.body.firstChild);
+        }
+    });
+
+    it('sends "next" and "previous" events on button click', () => {
```
[the remark this answers is not in the document]
Comment (satyasekharcvb) [reply force-app/main/default/lwc/paginator/__tests__/paginator.test.js]: Followed the paginator test sample from lwc-recipes. Re-implemented separate tests as suggested.

File: force-app/main/default/lwc/paginator/__tests__/paginator.test.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        // will automatically wait for the Promise chain to complete before
+        // ending the test and fail the test if the promise rejects.
+        return Promise.resolve().then(() => {
+            // Validate if mocked events got fired
+            expect(handlerPrevious.mock.calls.length).toBe(1);
+            expect(handlerNext.mock.calls.length).toBe(1);
+        });
+    });
+
+    it('displays total item count, page number, and number of pages', () => {
```
[the remark this answers is not in the document]
Comment (satyasekharcvb) [reply force-app/main/default/lwc/paginator/__tests__/paginator.test.js]: Implemented with changes

File: force-app/main/default/lwc/paginator/__tests__/paginator.test.js (hunk 1 above)
Comment (satyasekharcvb) [reply force-app/main/default/lwc/paginator/__tests__/paginator.test.js]: Implemented with changes

File: force-app/main/default/lwc/paginator/__tests__/paginator.test.js (hunk 2 above)
Comment (satyasekharcvb) [reply force-app/main/default/lwc/paginator/__tests__/paginator.test.js:26]: Implemented with changes

File: force-app/main/default/lwc/paginator/__tests__/paginator.test.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (pozil) [reply force-app/main/default/lwc/paginator/__tests__/paginator.test.js]: Ok, I see. It's a bit "light" so I'll update it after we're done with this one. Thx for the heads up.

File: force-app/main/default/lwc/paginator/__tests__/paginator.test.js (hunk 2 above)
Comment (satyasekharcvb) [reply force-app/main/default/lwc/paginator/__tests__/paginator.test.js:26]: This is somehow returning null. I will check and fix it to select only one element.

**Comment by pozil** [comment]
You can add a CSS class if needed.

**Pull request merged**